src/mpid/ch4/shm/xpmem/xpmem_init.c

```diff
@@ -50,12 +55,30 @@ int MPIDI_XPMEM_mpi_init_hook(int rank, int size, int *n_vcis_provided, int *tag
                                     (void **) &xpmem_segids, MPL_MEM_SHM);
     if (mpi_errno)
         MPIR_ERR_POP(mpi_errno);
+#ifdef MPIDI_CH4_SHM_XPMEM_COOP_P2P
+    mpi_errno = MPIDU_shm_seg_alloc(sizeof(uint64_t), (void **) &coop_caddr, MPL_MEM_SHM);
+    if (mpi_errno)
+        MPIR_ERR_POP(mpi_errno);
+#endif
     mpi_errno = MPIDU_shm_seg_commit(&shm_seg,
                                      &shm_seg_barrier,
                                      num_local, my_local_rank, local_rank_0, rank, MPL_MEM_SHM);
     if (mpi_errno)
         MPIR_ERR_POP(mpi_errno);
 
+#ifdef MPIDI_CH4_SHM_XPMEM_COOP_P2P
+    if (!my_local_rank) {
+        coop_counter =
+            (OPA_int_t *) MPL_malloc(num_local * num_local * sizeof(OPA_int_t), MPL_MEM_OTHER);
+        MPIR_ERR_CHKANDJUMP(coop_counter == NULL, mpi_errno, MPI_ERR_OTHER, "**nomem");
+
+        for (i = 0; i < num_local * num_local; ++i)
+            OPA_store_int(coop_counter + i, 0);
```

Comment:
Can `OPA_store_int` not handle array notation?

```suggestion
            OPA_store_int(&coop_counter[i], 0);
```